### DISCUSSION TITLE
Is it possible to determine whether `Option<T>` having a value within the debugger's variable tab? (or fetching inside values)

### DISCUSSION THREAD

**Discussion by ilsubyeega** [rust-lang/rust-analyzer#15462] [Q&A] [answered]
Is it possible to determine whether Option<T> having a value or fetching inside value within the debugger's variable tab? I can only get raw values. I'm not sure that this is possible or not.
![image](https://github.com/rust-lang/rust-analyzer/assets/37479424/cdbe4a23-d443-47c5-828c-0ab865020ef5)

**Comment by lnicola** [comment, marked as answer]
No, see https://github.com/rust-lang/rust-analyzer/issues/15407.

**Reply by ilsubyeega** [reply]
Thanks you for answering, now i can know debugging was out-of-scope of rust-analyzer.

**Comment by lnicola** [comment]
By the way, this seems to work fine on my platform, so you might want to try a different setup:

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/239e06db-056e-4a51-8671-ebcceb627302)

**Reply by ilsubyeega** [reply]
At least it was possible to unwrap the Some values for finding inner values, but not None through
![image](https://github.com/rust-lang/rust-analyzer/assets/37479424/90b729e6-1ae3-4d8c-b3f4-3efc2821d5e3)

Well seems debugging in WSL works, i will try to test another setup at Windows, thanks.
![image](https://github.com/rust-lang/rust-analyzer/assets/37479424/90e00f62-e96c-419e-844b-d8770857618b)
